src/tree/Treemap.js

```diff
@@ -0,0 +1,73 @@
+﻿"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "css!./Treemap"], factory);
+    } else {
+        root.tree_Treemap = factory(root.d3, root.common_HTMLWidget);
+    }
+}(this, function (d3, HTMLWidget) {
+    function Treemap(target) {
+        HTMLWidget.call(this);
+        this._tag = "div";
+    }
+    Treemap.prototype = Object.create(HTMLWidget.prototype);
+    Treemap.prototype.constructor = Treemap;
+    Treemap.prototype._class += " tree_Treemap";
+
+    Treemap.prototype.publish("stringProp", "defaultValue", "string", "Sample Property");
+
+    Treemap.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        this.myroot = element.append("div");
+    };
+
+    Treemap.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        var width = this.width();
+        var height = this.height();
+        var color = d3.scale.category20c();
```

Comment:
Search for ```_palette``` and ```paletteID``` to see how we use the global palette pool.

But essentially you want:
```
Treemap.prototype._palette = Palette.ordinal("default");
Treemap.prototype.publish("paletteID", "default", "set", "Palette ID", Treemap.prototype._palette.switch());
```

Then near the top of your update method:
```
        this._palette = this._palette.switch(this.paletteID());
```

Finally search/replace ```color``` with ```this._palette```